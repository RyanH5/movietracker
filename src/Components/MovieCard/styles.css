button {
  font-family: 'Monoton';
  font-style: italic;
  padding: 7px;
  color: cadetblue;
  border-radius: 7px;
}

.movie-card {
  border-radius: 10px;
  border: 4px solid darkcyan;
  width: 24%;
  height: 300px;
  overflow: hidden;
  background-color: aliceblue;
}

.mydiv {
  height: 30px;
  width: 100px;
}