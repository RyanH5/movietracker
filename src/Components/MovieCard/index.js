import React from 'react';
import './styles.css';
import PropTypes from 'prop-types';
import posterPlaceholder from './images/posterPlaceholder.jpg';
import { connect } from 'react-redux';
import { addFavorite, removeFromFavorites } from '../../Actions';
import { postFavorite, removeFaveFromDatabase } from '../../apiCalls';
import { withRouter } from 'react-router';

const MovieCard = (props) => {
  const { title, voteAverage, poster, overview, popularity, id, releaseDate } = props;
  const userId = props.userId;
  const movie = { id, userId, title, poster, releaseDate, voteAverage, overview };
  const pathAddition = 'favorites/new';
  
  const renderImage = () => {
    return poster ? <img
      src={`https://image.tmdb.org/t/p/w200${poster}`}
      alt="" /> :

      <img src={posterPlaceholder} />;
  };
  
  const isDuplicate = (id) => {
    const nonDuplicates = props.favorites.filter((fav) => {
      return fav.id !== id;
    });
    
    return props.favorites.length !== nonDuplicates.length;
  };
  
  const handleFavorite = async (id) => {
    const pathDeletion = `${userId}/favorites/${id}`;
    if (props.isLoggedIn) {
      if (isDuplicate(id)){
        
        props.removeFromFavorites(id);
        await removeFaveFromDatabase(pathDeletion);
      } else {
        await props.addFavorite(movie);
        
        postFavorite(pathAddition, movie, props.state.user);
      }
    } else {
      props.history.push('/login');
    }
  };

  return (
    <div className="movie-card"> 
      <button
        className="fave-button"
        onClick={() => handleFavorite(id)}>
        FAVORITE
      </button>
      <h1 className="card-title">{title}</h1>
      
      {renderImage()}
      <h4>Vote average:  {voteAverage}</h4>
      <h4>Popularity:  {popularity}</h4>
      {overview && <h5 className="overview">Summary:   {overview}</h5>}
    </div>
  );
};

MovieCard.propTypes = {
  title: PropTypes.string,
  id: PropTypes.number,
  voteAverage: PropTypes.number,
  poster: PropTypes.string,
  overview: PropTypes.string,
  popularity: PropTypes.number,
  backdrop: PropTypes.string,
  releaseDate: PropTypes.string,
  addFavorite: PropTypes.func,
  state: PropTypes.object,
  isLoggedIn: PropTypes.bool,
  removeFromFavorites: PropTypes.func,
  favorites: PropTypes.array,
  history: PropTypes.object
};

export const mapStateToProps = (state) => ({
  state,
  isLoggedIn: state.user.loginStatus,
  userId: state.user.id,
  favorites: state.user.favorites
});

export const mapDispatchToProps = (dispatch) => ({
  addFavorite: (movie) => dispatch(addFavorite(movie)),
  removeFromFavorites: (id) => dispatch(removeFromFavorites(id))
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(MovieCard));