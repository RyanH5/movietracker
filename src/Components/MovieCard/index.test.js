import React from 'react';
import { shallow } from 'enzyme';
import MovieCard from './index';

describe('MovieCard', () => {
  it('should match snapshot', () => {
    let wrapper = shallow(<MovieCard />);
    expect(wrapper).toMatchSnapshot();
  });
});