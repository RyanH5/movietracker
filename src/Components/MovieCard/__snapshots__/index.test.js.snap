// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieCard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button>
          FAVORITE
</button>,
        <h1 />,
        <h3 />,
        <h4 />,
        <h5 />,
        <img
          alt=""
          src={undefined}
/>,
      ],
      "className": "movie-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "FAVORITE",
        },
        "ref": null,
        "rendered": "FAVORITE",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "",
          "src": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button>
            FAVORITE
</button>,
          <h1 />,
          <h3 />,
          <h4 />,
          <h5 />,
          <img
            alt=""
            src={undefined}
/>,
        ],
        "className": "movie-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "FAVORITE",
          },
          "ref": null,
          "rendered": "FAVORITE",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
